### PR TITLE
[Tedix] Release 2020-06-15

### PR DESCRIPTION
**Pull request by sendr**
[T-318](https://youtrack.raccoongang.com/issue/T-318) -  `Order of the Past fields`
[T-312](https://youtrack.raccoongang.com/issue/T-312) -  `Align blocks  on the Manage Courses page`
[T-292 ](https://youtrack.raccoongang.com/issue/T-292) -  `To change color of messages to ''Tedix'' green`
[T-282](https://youtrack.raccoongang.com/issue/T-282) -  `order the Student List alphabetically.`
[T-298](https://youtrack.raccoongang.com/issue/T-298) -  `To make the columns wider`
[T-303](https://youtrack.raccoongang.com/issue/T-303) -  `To highlight the Header of the Tables and rows Future and Past`
[T-293](https://youtrack.raccoongang.com/issue/T-293) -  `To add the label`
[T-311](https://youtrack.raccoongang.com/issue/T-311) -  `[Report pop up] Student’s name isn't moved to the second line`
[T-287](https://youtrack.raccoongang.com/issue/T-287) -  `Fix filters behavior`

Release notes:
https://raccoongang.atlassian.net/wiki/spaces/tech/pages/1013481722/Tedix+Dev+info

### CODE REVIEW

File: lms/templates/manage_courses.html [hunk 1]
```diff
@@ -111,24 +120,57 @@ <h1>Manage Courses</h1>
...
 
         $(document).on('keyup', '#courseFilter', {targetList: $coursesList}, filter);
         $(document).on('keyup', '#classroomFilter', {targetList: $classroomList}, filter);
         $(document).on('keyup', '#studentFilter', studentFilter);
         $(document).on('change', '#id_classrooms', studentFilter);
 
+
+        function keywordMatches(keyword, text) {
+            var template = '';
+            for(var i=0; i < keyword.length; i++) {
```
Comment (cmltaWt0) [inline lms/templates/manage_courses.html:158]:
```suggestion
            for (var i=0; i < keyword.length; i++) {
```

File: lms/templates/manage_courses.html [hunk 2]
```diff
@@ -111,24 +120,57 @@ <h1>Manage Courses</h1>
...
+            ț: '([tţț]{1})',
+        };
 
         $(document).on('keyup', '#courseFilter', {targetList: $coursesList}, filter);
         $(document).on('keyup', '#classroomFilter', {targetList: $classroomList}, filter);
         $(document).on('keyup', '#studentFilter', studentFilter);
         $(document).on('change', '#id_classrooms', studentFilter);
 
+
+        function keywordMatches(keyword, text) {
```
Comment (cmltaWt0) [inline lms/templates/manage_courses.html:156]: Add comment please.

File: lms/templates/manage_courses.html [hunk 3]
```diff
@@ -111,24 +120,57 @@ <h1>Manage Courses</h1>
...
+            return option1Text.localeCompare(option2Text);
+        });
+        options.appendTo("#id_students");
+
         $('.manage-courses-wrapper select, .manage-courses-wrapper input[type="checkbox"]').styler();
 
         var $coursesList = $("#id_courses-styler li");
         var $classroomList = $("#id_classrooms-styler li");
         var $studentList = $("#id_students-styler li");
+        var characters = {
```
Comment (cmltaWt0) [inline lms/templates/manage_courses.html:136]: Add comment

File: common/djangoapps/tedix_ro/views.py [hunk 4]
```diff
@@ -102,34 +103,28 @@ def user_due_date_data(user, due_date_datetime):
...
+            displayed_date = '{}: {}'.format(date_group, date.strftime('%d %B'))
         elif (today - due_date).days > 0:
             date_group = 'Past'
-            due_date_data.update({
-                'displayed_date': '{}: {}'.format(date_group, date.strftime('%d %B')),
-                'due_date_order': 4
-            })
+            due_date_order = 4
+            displayed_date = '{}: {}'.format(date_group, date.strftime('%d %B'))
+            date_data *= -1
```
Comment (cmltaWt0) [inline common/djangoapps/tedix_ro/views.py:121]: Add comment for this.

File: common/djangoapps/tedix_ro/views.py [hunk 5]
```diff
@@ -102,34 +103,28 @@ def user_due_date_data(user, due_date_datetime):
...
         due_date = due_date_datetime.date()
         today = utc_now.date()
+        date_data = time.mktime(date.timetuple())
         if today == due_date:
-            date_group = 'Today'
-            due_date_data.update({
-                'displayed_date': date_group,
-                'due_date_order': 1
-            })
+            displayed_date = date_group = 'Today'
```
Comment (cmltaWt0) [inline common/djangoapps/tedix_ro/views.py:108]: Do we need to translate this?

File: common/djangoapps/tedix_ro/views.py [hunk 6]
```diff
@@ -102,34 +103,28 @@ def user_due_date_data(user, due_date_datetime):
...
+            due_date_order = 1
         elif (today - due_date).days == -1:
-            date_group = 'Tomorrow'
-            due_date_data.update({
-                'displayed_date': date_group,
-                'due_date_order': 2
-            })
+            displayed_date = date_group = 'Tomorrow'
```
Comment (cmltaWt0) [inline common/djangoapps/tedix_ro/views.py:111]: The same as above.

File: lms/templates/manage_courses.html [hunk 7]
```diff
@@ -18,38 +18,47 @@
 <%block name="pagetitle">${_("Manage Courses")}</%block>
 
 <main id="main">
-    <h1>Manage Courses</h1>
+    <h1 class="main-title">
+        Manage Courses
+        <span class="title-tooltip">
+            Keep the CTRL (Command ⌘ for Mac) key pressed and CLICK on the items to select multiple entries.
```
Comment (cmltaWt0) [inline lms/templates/manage_courses.html:24]: Do we need to translate this?

File: lms/templates/manage_courses.html [hunk 8]
```diff
@@ -18,38 +18,47 @@
...
         <form method="post">
             <input type="hidden" name="csrfmiddlewaretoken" value="${ csrftoken }" />
-            <div class="courses-field field courses-list">
-                <div class="head">
-                    <label>Filter Courses:</label>
-                    <input id="courseFilter" type="text">
+            <div class="courses-field-wrapper">
+                <div class="courses-field field courses-list">
+                    <div class="head">
+                        <label>Filter Courses:</label>
```
Comment (cmltaWt0) [inline lms/templates/manage_courses.html:34]: Translation?

File: lms/templates/manage_courses.html [hunk 9]
```diff
@@ -111,24 +120,57 @@ <h1>Manage Courses</h1>
             format: 'd/m/Y H:i',
         });
 
+        var options = $("#id_students option");
+        options.detach().sort(function(option1, option2) {
```
Comment (cmltaWt0) [inline lms/templates/manage_courses.html:124]: Add comment for this detach->sort.

File: lms/templates/manage_courses.html [hunk 10]
```diff
@@ -111,24 +120,57 @@ <h1>Manage Courses</h1>
...
+
+        function keywordMatches(keyword, text) {
+            var template = '';
+            for(var i=0; i < keyword.length; i++) {
+                template += characters[keyword[i]] || keyword[i];
+            }
+            template += '.*';
+
+            var reg = new RegExp(template);
+            return (text.match(reg));
```
Comment (cmltaWt0) [inline lms/templates/manage_courses.html:164]: Why do we need this additional parentheses?
